### PR TITLE
feat: add named provider metadata accessor

### LINKED ISSUE #714: [FEATURE] Named provider metadata accessor
### Requirements

Add support for accessing provider metadata for named providers. If a name is invalid, the default provider metadata should be returned.

### PR DESCRIPTION
**Pull request by beeme1mr**
## This PR

- adds an accessor for named provider metadata

### Related Issues

Fixes #714 

### Notes

The default metadata is returned if the name doesn't match a registered provider.

This is another example where provider namespace would work well. It doesn't make sense to call this "named client" since we are only interested in the client.

### CODE REVIEW

File: packages/client/test/open-feature.spec.ts [hunk 1]
```diff
@@ -39,13 +39,13 @@ describe('OpenFeature', () => {
     describe('Requirement 1.1.2.1', () => {
       it('should throw because the provider is not intended for the client', () => {
         const provider = mockProvider({ runsOn: 'server' });
-        expect(() => OpenFeature.setProvider(provider)).toThrowError(
-          "Provider 'mock-events-success' is intended for use on the server."
+        expect(() => OpenFeature.setProvider(provider)).toThrow(
```
Comment (beeme1mr) [inline packages/client/test/open-feature.spec.ts:42]: I changed this because `toThrowError` has been deprecated and I was already modifying this file.

Comment (toddbaert) [reply packages/client/test/open-feature.spec.ts:42]: Was just about to ask. Thanks!

File: packages/client/test/open-feature.spec.ts [hunk 2]
```diff
@@ -132,6 +132,22 @@ describe('OpenFeature', () => {
       OpenFeature.setProvider('client2', { ...provider1 });
       expect(provider1.onClose).toHaveBeenCalledTimes(1);
     });
+
+    it('should return the default provider metadata when passing an invalid client name', async () => {
```
Comment (toddbaert) [inline packages/client/test/open-feature.spec.ts:136]:
```suggestion
    it('should return the default provider metadata when passing an unregistered client name', async () => {
```

File: packages/client/test/open-feature.spec.ts [hunk 3]
```diff
@@ -132,6 +132,22 @@ describe('OpenFeature', () => {
...
+      const mockProvider = { metadata: { name: 'test' } } as unknown as Provider;
+      OpenFeature.setProvider(mockProvider);
+      const metadata = OpenFeature.getProviderMetadata('invalid');
+      expect(metadata.name === mockProvider.metadata.name).toBeTruthy();
+    });
+
+    it('should return the named provider metadata when passing a valid client name', async () => {
```
Comment (toddbaert) [inline packages/client/test/open-feature.spec.ts:143]:
```suggestion
    it('should return the named provider metadata when passing a registered client name', async () => {
```

File: packages/server/test/open-feature.spec.ts [hunk 4]
```diff
@@ -132,6 +132,22 @@ describe('OpenFeature', () => {
       OpenFeature.setProvider('client2', { ...provider1 });
       expect(provider1.onClose).toHaveBeenCalledTimes(1);
     });
+
+    it('should return the default provider metadata when passing an invalid client name', async () => {
```
Comment (toddbaert) [inline packages/server/test/open-feature.spec.ts:136]:
```suggestion
    it('should return the default provider metadata when passing an unregistered client name', async () => {
```

File: packages/server/test/open-feature.spec.ts [hunk 5]
```diff
@@ -132,6 +132,22 @@ describe('OpenFeature', () => {
...
+      const mockProvider = { metadata: { name: 'test' } } as unknown as Provider;
+      OpenFeature.setProvider(mockProvider);
+      const metadata = OpenFeature.getProviderMetadata('invalid');
+      expect(metadata.name === mockProvider.metadata.name).toBeTruthy();
+    });
+
+    it('should return the named provider metadata when passing a valid client name', async () => {
```
Comment (toddbaert) [inline packages/server/test/open-feature.spec.ts:143]:
```suggestion
    it('should return the named provider metadata when passing a registered client name', async () => {
```

File: packages/server/test/open-feature.spec.ts [hunk 6]
```diff
@@ -132,6 +132,22 @@ describe('OpenFeature', () => {
       OpenFeature.setProvider('client2', { ...provider1 });
       expect(provider1.onClose).toHaveBeenCalledTimes(1);
     });
+
+    it('should return the default provider metadata when passing an invalid client name', async () => {
+      const mockProvider = { metadata: { name: 'test' } } as unknown as Provider;
+      OpenFeature.setProvider(mockProvider);
+      const metadata = OpenFeature.getProviderMetadata('invalid');
```
Comment (toddbaert) [inline packages/server/test/open-feature.spec.ts:139]:
```suggestion
      const metadata = OpenFeature.getProviderMetadata('unused');
```

File: packages/client/test/open-feature.spec.ts [hunk 7]
```diff
@@ -132,6 +132,22 @@ describe('OpenFeature', () => {
       OpenFeature.setProvider('client2', { ...provider1 });
       expect(provider1.onClose).toHaveBeenCalledTimes(1);
     });
+
+    it('should return the default provider metadata when passing an invalid client name', async () => {
+      const mockProvider = { metadata: { name: 'test' } } as unknown as Provider;
+      OpenFeature.setProvider(mockProvider);
+      const metadata = OpenFeature.getProviderMetadata('invalid');
```
Comment (toddbaert) [inline packages/client/test/open-feature.spec.ts:139]:
```suggestion
      const metadata = OpenFeature.getProviderMetadata('unused');
```

File: packages/shared/src/open-feature.ts [hunk 8]
```diff
@@ -50,11 +50,21 @@ export abstract class OpenFeatureCommonAPI<P extends CommonProvider = CommonProv
...
    * @returns {ProviderMetadata} Provider Metadata
    */
   get providerMetadata(): ProviderMetadata {
-    return this._defaultProvider.metadata;
+    return this.getProviderMetadata();
+  }
+
+  /**
+   * Get metadata about a registered provider using the client name.
+   * An invalid or empty client name will return the default provider.
```
Comment (toddbaert) [inline packages/shared/src/open-feature.ts:62]:
```suggestion
   * An unbound or empty client name will return the default provider.
```

**Review by toddbaert — APPROVED** [review]
Just terminology recommendations.

File: packages/shared/src/open-feature.ts [hunk 9]
```diff
@@ -50,11 +50,21 @@ export abstract class OpenFeatureCommonAPI<P extends CommonProvider = CommonProv
   }
 
   /**
-   * Get metadata about registered provider.
+   * Get metadata about default provider.
```
Comment (beeme1mr) [inline packages/shared/src/open-feature.ts:53]:
```suggestion
   * Get metadata about the default provider.
```

File: packages/shared/src/open-feature.ts [hunk 10]
```diff
@@ -50,11 +50,21 @@ export abstract class OpenFeatureCommonAPI<P extends CommonProvider = CommonProv
...
    * @returns {ProviderMetadata} Provider Metadata
    */
   get providerMetadata(): ProviderMetadata {
-    return this._defaultProvider.metadata;
+    return this.getProviderMetadata();
+  }
+
+  /**
+   * Get metadata about a registered provider using the client name.
+   * An unbound or empty client name will return the default provider.
```
Comment (beeme1mr) [inline packages/shared/src/open-feature.ts:62]:
```suggestion
   * An unbound or empty client name will return metadata from the default provider.
```

**Review by lukas-reining — APPROVED** [review]
Looks good!

**Pull request merged**